src/lib/walletStation.js

```diff
@@ -15,133 +13,224 @@ export const broadcastModeObj = {
     BROADCAST_MODE_UNSPECIFIED: "BROADCAST_MODE_UNSPECIFIED",
 };
 
+export const gasDefault = { gas: "200000", amount: '0' }
+
 export default class WalletStation {
-    constructor() {
-        const cosmos = new Cosmos(network.lcd, network.chainId);
-        cosmos.setBech32MainPrefix(consts.DENOM);
-        this.cosmos = cosmos;
-    }
+    constructor() { }
+    collectWallet = async () => {
+        const keplr = await window.Keplr.getKeplr();
+        if (!keplr) {
+            throw consts.INSTALL_KEPLR_FIRST;
+        }
+        return await keplr.getOfflineSignerAuto(network.chainId);
+    };
 
-    constructTxBody = async (messages) => {
-        const block = await this.cosmos.get('/blocks/latest');
-        const timeoutHeight = parseInt(block.block.header.height) + consts.NUM.TX_TIMEOUT_HEIGHT;
-        return this.cosmos.constructTxBody({ messages, timeout_height: timeoutHeight });
-    }
+    signerClient = async (wallet) => {
+        return await cosmwasm.SigningCosmWasmClient.connectWithSigner(network.rpc, wallet, {
+            gasPrice: new GasPrice(Decimal.fromUserInput('0', 6), network.denom),
+            prefix: network.denom,
+            gasLimits: { exec: 10000000 },
```

Comment:
I think this line is redundant, because the library does not have the gasLimits option

Reply:
Thanks !